x/auth/ante/basic.go

```diff
@@ -100,10 +101,15 @@ func (cgts ConsumeTxSizeGasDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, sim
 	// simulate gas cost for signatures in simulate mode
 	if simulate {
 		// in simulate mode, each element should be a nil signature
-		sigs := sigTx.GetSignatures()
+		sigs, err := sigTx.GetSignaturesV2()
+		if err != nil {
+			return ctx, err
+		}
+		n := len(sigs)
```

Comment:
Since you only use the length once, why put it in a variable?

Reply:
It’s used in the for loop below for every signer. I prefer not to call length in a for loop.

Reply:
Why? It's free.

Reply:
Because l used to be a C/C++ programmer 🤷🏻‍♂️

Reply:
lol fair -- not blocking on this, just a nit to only use a var if its used more than once.